x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index_check_flyout/constants.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const DEFAULT_HISTORICAL_RESULTS_START_DATE = 'now-30d';
+export const DEFAULT_HISTORICAL_RESULTS_END_DATE = 'now';
+
+export const ALL_TAB_ID = 'allTab' as const;
+export const ECS_COMPLIANT_TAB_ID = 'ecsCompliantTab' as const;
+export const CUSTOM_TAB_ID = 'customTab' as const;
+export const INCOMPATIBLE_TAB_ID = 'incompatibleTab' as const;
+export const SAME_FAMILY_TAB_ID = 'sameFamilyTab' as const;
```

Comment:
NIT: we could use `enum`, so we have the types for free.

Reply:
Yes, enums have been discussed many times in Elastic Slack threads. I can not find them now, but the current convention is that it is fine to use them. The only use case that may cause performance issues is when trying to iterate over all the values, which is not the case here. 
The example of "type intercompatibility" is not a good example because it is showcasing exactly what enums try to prevent, hardcoded constants. The real issue happens when a component outside our control uses `string` and we need cast, but that is also a problem when using `as const` as in this code.

No problem though, this is only a NIT, no need to change it.

Reply:
I should be able to pass string literals in tests for example without having to import the enum, because enum thinks it's not the same string when it IS.

Also javascript engines use string interning (https://stackoverflow.com/a/5276975/2998898, coming from a compiler engineer). Which literally leaves 2 reasons to not use strings:

1. Because strings can be accidentally mutated without people noticing (only true in JS, not true in TS because as const narrows the string type down)
2. Magic strings are bad. Again not the case under discussion here because if we refer to string that are (as const)-ed they are already assigned to a variable which means they are no longer magical.

So I'd argue type intercompatibility is a pretty good example.

> The real issue happens when a component outside our control uses string and we need cast, but that is also a problem when using as const as in this code.

Didn't get this part tho. Maybe let's discuss sync.

Let me know if I can resolve this.

Reply:
Yes, we both agree not to use strings, good, but I am not arguing we should...

> I should be able to pass string literals in tests for example without having to import the enum

I don't get why you should be able to do that

The type intercompatibility thing that you consider a problem is actually a feature. What enums do is abstract the actual value, it makes it opaque to the outside and it forces the use of the enum over string literals. This will come in handy if you ever have to do a refactor.

Why I prefer `enum` over `as const` variables: 
- The code is more maintainable (value abstraction + we don't need to maintain the types boilerplate).
- The code is more readable (we never lose context since it does not allow using string literals).
- It makes the code easier to work with (better IntelliSense)

Nevertheless, I have no problem keeping the current approach, it is also fine, I will not block the PR for this. Feel free to resolve this topic if you want. and we can discuss it sync.

Reply:
@semd

I agree about the types boilerplate.
I agree about ease of refactor.

I disagree about intellisense because TS infers string literals as good as enums and it's available in all editors on lsp level
I disagree about more readable, I think it's about the same if you factor in that the only time you will use literals against this variable is against it's type when accepting passing the literal into function. And since it will already be typed you will get the same inference from your editor, especially if you have inlays enabled. So on purely technical level you are right but if you factor in widely accepted editor support I'd argue the it's more equal than not.

Good talks, thanks for your perspective.